### PR TITLE
Fixed: Order history pagination (#4599)

### LINKED ISSUE #4599: Order History Pagination Not Working Properly
## Current behavior

**Issue** : While viewing order history, the user can not see more than 20 orders.

**Why its not working** : lazyLoadOrdersOnScroll implemented on the order history page, calls getOrdersHistory,
which previously used to fetch all the orders, but on introducing pagination support in "getOrdersHistory", it only fetches 20 orders, which is its default value. 
Still, the order history page calls it only once, trying to fetch all the results, and on those results, it applies lazyLoadOrdersOnScroll.

Due to which the user can not see more than 20 orders. 

## Expected behavior

User should be able to see all the orders while scrolling through the order history page.

## Steps to reproduce the issue

1. Login with a valid user having more than 20 orders.
1. Go to profile.
1. Click on order history.
1. Scroll down to view more orders.

## Can you handle fixing this bug by yourself?
* [ ] YES
* [ ] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

* [x] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
* [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
* [x] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

* Browser: 
* OS: 
* Node: 
* Code Version:

### PR DESCRIPTION
**Pull request by AishwaryShrivastav**
### Related Issues

closes #4599 

### Short Description and Why It's Useful

the getOrdersHistory, which previously used to fetch all the orders, after introducing pagination support in "getOrdersHistory", it only fetches 20 orders, which is its default value.

Still, the order history page calls it only once, trying to fetch all the results, and on those results, it applies lazyLoadOrdersOnScroll.

Due to which the user can not see more than 20 orders.

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature
- [ ] (Next only) I've followed this [instruction](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and I've created a `.js` file with information about my Pull Request

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by AishwaryShrivastav** [comment]
Created a new action for user state to append orders in the state, which will only get calls from pagination and remove the previously added appending logic from refreshOrderHistory.
We had added appending logic for pagination on getOrderHistory, it fetches order history and appends to the state, if state already had any orders, but it created these issues:

1. On every window refresh, getOrderHistory runs and appends the same orders again into the state. 

2. On placing the order,  getOrderHistory runs and appends the new order again into the state.  


So, For pagination, we wrote another action only append orders and will only run from pagination and we removed the changes from getOrderHistory ( we added to append the state, in case of older orders in state) so that older action will always refresh the state as we wrote a new action which will only append the state and only get call from pagination.

File: core/modules/order/components/UserOrdersHistory.ts [hunk 1]
```diff
@@ -26,6 +27,14 @@ export default {
   methods: {
     onBottomScroll () {
       ++this.pagination.current
-    }
+      const totalCount = this.$store.state.user.orders_history.totalCount ? this.$store.state.user.orders_history.totalCount : 0
```
Comment (lukaszjedrasik) [inline core/modules/order/components/UserOrdersHistory.ts:30]: Attribute name is `total_count`, not `totalCount`
```js
this.$store.state.user.orders_history.total_count ? this.$store.state.user.orders_history.total_count : 0
```

File: core/modules/order/components/UserOrdersHistory.ts [hunk 2]
```diff
@@ -26,6 +27,14 @@ export default {
...
+      const isLastPage = this.pagination.current > Math.ceil(totalCount / this.pagination.perPage);
+      if (!isLastPage) {
+        this.appendNewOrders({ currentPage: this.pagination.current, refresh: true })
```
Comment (lukaszjedrasik) [inline core/modules/order/components/UserOrdersHistory.ts:33]: 1. I think that `refresh: true` is unnecessary.  
2. Argument type `{ refresh: boolean, currentPage: any }` is not assignable to parameter type `{ resolvedFromCache: any, pageSize?: any, currentPage?: any }` 

Zrzut ekranu 2021-04-22 o 12 07 38

File: core/modules/user/store/actions.ts [hunk 3]
```diff
@@ -224,6 +224,7 @@ const actions: ActionTree<UserState, RootState> = {
     commit(types.USER_GROUP_TOKEN_CHANGED, '')
     commit(types.USER_GROUP_CHANGED, null)
     commit(types.USER_INFO_LOADED, null)
+    commit(types.USER_ORDERS_HISTORY_LOADED, null)
```
Comment (lukaszjedrasik) [inline core/modules/user/store/actions.ts:227]: It is duplicated. Look at line `233`.

**Review by lukaszjedrasik — CHANGES_REQUESTED** [review]
I think that problem still occurs: https://github.com/vuestorefront/vue-storefront/pull/4632#issuecomment-692658418

So please do a lot of manual tests.

**Comment by lukaszjedrasik** [comment]
New PR: https://github.com/vuestorefront/vue-storefront/pull/5850

**Pull request closed without merging**